frontend/app/src/scenes/Home/scenes/UpsertCategory/index.tsx

```diff
@@ -139,7 +203,92 @@ class UpsertCategory extends React.Component<
+                                  className={mainStyles.closeButton}
+                                  onClick={() => this.handleClose(language)}
+                                >
+                                  <DeleteOutlined />
+                                </Button>
+                                <Title level={3}>{translationMeta.name}</Title>
+                                <Text>{translationMeta.title}</Text>
+                                <Input
+                                  className={mainStyles.formItem}
+                                  value={translation.name}
+                                  placeholder={translationMeta.placeholder}
+                                  onChange={(event) =>
+                                    this.handleTranslations(
+                                      language,
+                                      event.target.value
+                                    )
+                                  }
+                                />
+                              </Card>
+                            );
+                          }
+                        }
+                      )}
+                      {!this.state.isAddTranslationSelected && (
+                        <>
+                          <Title level={4}>Translation of the Category</Title>
+                          <Select
+                            placeholder="Please add a translation"
+                            className={mainStyles.formSelect}
+                            onSelect={(value: string) =>
+                              this.handleLanguageSelect(value)
+                            }
+                          >
+                            {Array.from(translationMetaData.keys()).map(
+                              (language) => {
+                                const metaData = translationMetaData.get(
+                                  language
+                                );
+                                if (metaData) {
+                                  return (
+                                    <Option
+                                      key={language}
+                                      value={language}
+                                      className={mainStyles.formSelect}
+                                      disabled={this.state.translations.has(
+                                        language
+                                      )}
+                                    >
+                                      {metaData.nameInEnglish}
+                                    </Option>
+                                  );
+                                }
+                              }
+                            )}
+                          </Select>
+                        </>
+                      )}
+                      {this.state.isAddTranslationSelected &&
+                        this.state.translations.size != 2 && (
```

Comment:
not '2' it should be the length of language meta map

Reply:
ah yes